traits/trait_types.py

```diff
@@ -4001,15 +4000,25 @@ def __init__(self, *traits, **metadata):
                 "'default'."
             )
 
-        default_value = None
         if 'default_value' in metadata:
             default_value = metadata.pop("default_value")
-        elif self.list_ctrait_instances:
```

Comment:
The condition here is unnecessary: `list_ctrait_instances` is always nonempty.

Reply:
just to be clear, this is always nonempty because we can't/don't use `trait = Union()`. right?

Reply:
`Union()` is legal, but it's equivalent to `Union(None)`. That doesn't feel quite right to me (I think I'd prefer an exception), but fixing it is outside the scope of this issue. Code ref: https://github.com/enthought/traits/blob/385f42b3b8ccae576714227f4e40ed42c4c3d149/traits/trait_types.py#L3982-L3983